dsp-shared/src/main/scala/dsp/valueobjects/LangString.scala

```diff
@@ -0,0 +1,79 @@
+package dsp.valueobjects
+
+import dsp.errors.ValidationException
+import zio.prelude.Validation
+
+/**
+ * LangString value object
+ */
+sealed abstract case class LangString private (language: LanguageCode, value: String)
+
+object LangString {
+
+  /**
+   * Creates a [[zio.prelude.Validation]] that either fails with a ValidationException or succeeds with a LangString value object.
+   *
+   * @param language a [[LanguageCode]] value object representing the language of the LangString.
+   * @param value    the string value of the LangString.
+   * @return a Validation of a LangString value object.
+   */
+  def make(language: LanguageCode, value: String): Validation[ValidationException, LangString] =
+    if (value.isBlank()) {
+      Validation.fail(ValidationException(LangStringErrorMessages.LangStringValueEmpty))
+    } else {
+      Validation.succeed(new LangString(language, value) {})
+    }
+
+  /**
+   * Creates a [[zio.prelude.Validation]] that either fails with a ValidationException or succeeds with a LangString value object.
+   *
+   * @param language a two-digit language code string representing the language of the LangString.
+   * @param value    the string value of the LangString.
+   * @return a Validation of a LangString value object.
+   */
+  def makeFromStrings(language: String, value: String): Validation[ValidationException, LangString] =
+    for {
+      languageCode <- LanguageCode.make(language)
+      langString   <- LangString.make(languageCode, value)
+    } yield langString
+
+  /**
+   * Unsafely creates a LangString value object.
+   * Warning: skipps all validation. Should not be used unless there is no possibility for the data to be invalid.
+   *
+   * @param languagea [[LanguageCode]] value object representing the language of the LangString.
+   * @param value     the string value of the LangString.
+   * @return a LanguageCode value object
+   */
+  def unsafeMake(language: LanguageCode, value: String): LangString =
+    new LangString(language, value) {}
+}
+
+/**
+ * LanguageCode value object.
+ */
+sealed abstract case class LanguageCode private (value: String)
+object LanguageCode { self =>
+  def make(value: String): Validation[ValidationException, LanguageCode] =
+    if (value.isEmpty) {
+      Validation.fail(ValidationException(LanguageCodeErrorMessages.LanguageCodeMissing))
+    } else if (!V2.SupportedLanguageCodes.contains(value)) {
+      Validation.fail(ValidationException(LanguageCodeErrorMessages.LanguageCodeInvalid))
+    } else {
+      Validation.succeed(new LanguageCode(value) {})
+    }
+
+  lazy val en: LanguageCode = new LanguageCode(V2.EN) {}
+  lazy val de: LanguageCode = new LanguageCode(V2.DE) {}
+  lazy val fr: LanguageCode = new LanguageCode(V2.FR) {}
+  lazy val it: LanguageCode = new LanguageCode(V2.IT) {}
+  lazy val rm: LanguageCode = new LanguageCode(V2.RM) {}
+}
```

Comment:
I find this part unnecessarily separated and whole `LangString` implementation kind of overcomplicated. How about [this](https://github.com/dasch-swiss/dsp-api/blob/DEV-1010-dsp-api-add-role-slice/dsp-shared/src/main/scala/dsp/valueobjects/Role.scala#L21)? It is from my ongoing branch and some changes are not yet pushed, but IMO looks cleaner.

Reply:
let's discuss this in the thursday meeting. I think we need a value object for LanguageCode, but it's of course up for discussion